packages/firestore/src/api/database.ts

```diff
@@ -420,7 +421,12 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     const persistenceKey = IndexedDbPersistence.buildStoragePrefix(
       this.makeDatabaseInfo()
     );
-    return IndexedDbPersistence.clearPersistence(persistenceKey);
+    const deferred = new Deferred<void>();
+    this._queue.enqueueAndForget(async () => {
+      await IndexedDbPersistence.clearPersistence(persistenceKey);
```

Comment:
If the underlying `clearPersistence` rejects its promise, `await` will throw an exception and this will cause the enqueued function to fail the AsyncQueue.

I'm starting to think we need a test for this :-).

There are two solutions:
  * surround your await with try/catch, and reject in the catch block
  * resolve the outer deferred with the Promise from clearPersistence